records/generator_test.go

```diff
@@ -208,6 +210,14 @@ func TestInsertState(t *testing.T) {
 		{rg.As, "slave.mesos.", []string{"1.2.3.10", "1.2.3.11", "1.2.3.12"}},
 		{rg.As, "some-box.chronoswithaspaceandmixe.mesos.", []string{"1.2.3.11"}}, // ensure we translate the framework name as well
 		{rg.As, "marathon.mesos.", []string{"1.2.3.11"}},
+		{rg.SRVs, "_big-dog._tcp.marathon.mesos.", []string{
+			"big-dog-4dfjd-0.marathon.mesos.:80",
+			"big-dog-4dfjd-0.marathon.mesos.:443",
+		}},
+		{rg.SRVs, "_big-dog2._tcp.marathon.mesos.", []string{
```

Comment:
I think we're talking past each other. In my mind when there's discovery
info if you want to preserve the old behavior then you need ctx.taskName =
task.DiscoveryInfo.Name before taskContextRecord. for the new behavior, set
ctx.taskName=spec(task.DiscoveryInfo.Name) and call taskContextRecord again.

On Thu, Feb 25, 2016 at 1:12 PM, Sargun Dhillon <notifications@github.com>
wrote:

> In records/generator_test.go
> <https://github.com/mesosphere/mesos-dns/pull/399#discussion_r54138033>:
>
> > @@ -208,6 +210,14 @@ func TestInsertState(t *testing.T) {
> >  		{rg.As, "slave.mesos.", []string{"1.2.3.10", "1.2.3.11", "1.2.3.12"}},
> >  		{rg.As, "some-box.chronoswithaspaceandmixe.mesos.", []string{"1.2.3.11"}}, // ensure we translate the framework name as well
> >  		{rg.As, "marathon.mesos.", []string{"1.2.3.11"}},
> > +		{rg.SRVs, "_big-dog._tcp.marathon.mesos.", []string{
> > +			"big-dog-4dfjd-0.marathon.mesos.:80",
> > +			"big-dog-4dfjd-0.marathon.mesos.:443",
> > +		}},
> > +		{rg.SRVs, "_big-dog2._tcp.marathon.mesos.", []string{
>
> The idea is to continue to provide the existing behaviour + new behaviour:
>
>
>         // LEGACY TODO: REMOVE
>         rg.taskContextRecord(ctx, task, f, domain, spec)
>         // LEGACY, TODO: REMOVE
>
>         ctx.taskName = spec(task.DiscoveryInfo.Name)
>         rg.taskContextRecord(ctx, task, f, domain, spec)
>
> So, when the Marathon team adds a bunch of DiscoveryInfo entries, the old
> ones still exist.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/399/files#r54138033>.
>